lib/compress/zstd_compress.c

```diff
@@ -209,6 +209,16 @@ static ZSTD_CCtx_params ZSTD_makeCCtxParamsFromCParams(
     /* should not matter, as all cParams are presumed properly defined */
     ZSTD_CCtxParams_init(&cctxParams, ZSTD_CLEVEL_DEFAULT);
     cctxParams.cParams = cParams;
+
+    if (cParams.strategy >= ZSTD_btopt && cParams.windowLog >= 27 && cctxParams.nbWorkers == 0) {
```

Comment:
We don't need to check `cctxParams.nbWorkers` here. It will either always be true (in this case), or always false, since we've just initialized them.

We should move this check `cParams.strategy >= ZSTD_btopt && cParams.windowLog >= 27` to a function, so we don't need to write it twice. Otherwise the two invocations could drift.